### PR TITLE
Phase statuses

### PR DESCRIPTION
**Pull request by jimmykodes**

### CODE REVIEW

File: src/apps/competitions/tests/test_phase_migration.py [hunk 1]
```diff
@@ -95,3 +121,48 @@ def test_all_submissions_migrated_before_changing_phase_status(self):
...
+            start=now() - timedelta(minutes=1),
+            end=now() + timedelta(minutes=5),
+        )
+        self.phase3 = PhaseFactory(
+            **base,
+            start=now() + timedelta(minutes=10),
+            end=now() + timedelta(minutes=15),
+        )
+
+    def test_phase_migration_updates_statuses(self):
```
Comment (BrendanMyers88) [inline src/apps/competitions/tests/test_phase_migration.py]: Should probably have a test where phase1 status is `Phase.CURRENT`, phase2 status is `Phase.NEXT` and phase3 status is checked. 

Also might be worth checking statuses on competitions where there are 5 or more phases?

File: src/apps/competitions/tasks.py [hunk 2]
```diff
@@ -434,7 +434,29 @@ def unpack_competition(competition_dataset_pk):
...
+                current_phase_index = current_phase[0]['index']
+                previous_index = current_phase_index - 1 if current_phase_index >= 1 else None
+                next_index = current_phase_index + 1 if current_phase_index < len(competition['phases']) - 1 else None
+                competition['phases'][current_phase_index]['status'] = Phase.CURRENT
+                if next_index:
+                    competition['phases'][next_index]['status'] = Phase.NEXT
+                if previous_index:
+                    competition['phases'][previous_index]['status'] = Phase.PREVIOUS
+            else:
+                competition['phases'][0]['status'] = Phase.NEXT
```
Comment (jimmykodes) [inline src/apps/competitions/tasks.py]: Need to verify phases start in the future before setting "next" value

File: src/apps/competitions/tasks.py [hunk 3]
```diff
@@ -686,20 +708,61 @@ def create_competition_dump(competition_pk, keys_instead_of_files=True):
...
+
+    Competition.objects.filter(
+        pk__in=Phase.objects.annotate(
+            running_subs=Count(Subquery(
+                Submission.objects.filter(
+                    created_by_migration=OuterRef('pk')
+                ).exclude(
+                    status__in=status_list
+                ).values('pk')
+            ))).filter(
```
[the remark this answers is not in the document]
Comment (jimmykodes) [reply src/apps/competitions/tasks.py]: Sorta cleaned this up. Still not great

**Comment by jimmykodes** [comment]
- [ ] set statuses on comp edit form save

**Comment by ckcollab** [comment]
Works beautifully editing phases and saving now, believe after conflict fix we can :shipit:

**Pull request merged**